src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SparseIntMap.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Text.RegularExpressions.Symbolic
+{
+    /// <summary>An insertion-ordered map that supports small int keys.</summary>
+    /// <remarks>Uses a sparse array of the same size as the space of keys for efficient lookups.</remarks>
+    internal class SparseIntMap<T> where T : struct
+    {
+        private const int InitialCapacity = 16;
+        private List<(int, T)> _dense = new();
```

Comment:
```suggestion
        private List<KeyValuePair<int, T>> _dense = new();
```
Alternatively we could name the tuple elements Key and Value, but that's what KeyValuePair is for :) and reading Key and Value is a lot nicer than Item1 and Item2.